soco/core.py

```diff
@@ -1490,6 +1494,70 @@ def get_item_album_art_uri(self, item):
         else:
             return None
 
+    @only_on_master
+    def set_sleep_timer(self, sleep_time_seconds):
+        """Sets the sleep timer.
+
+        Args:
+            sleep_time_seconds (`int/NoneType`): How long to wait before
+                turning off speaker in seconds, None to cancel a sleep timer.
+                Maximum value of 86399
+
+        Raises:
+            SoCoException: Upon errors interacting with Sonos controller
+            ValueError: Argument/Syntax errors
+
+        """
+        # Note: A value of None for sleep_time_seconds is valid, and needs to
+        # be preserved distinctly separate from 0. 0 means go to sleep now,
+        # which will immediately start the sound tappering, and could be a
+        # useful feature, while None means cancel the current timer
+        if sleep_time_seconds == 0:
+            sleep_time = '00:00:00'
+        elif not sleep_time_seconds:
+            sleep_time = ''
+        elif sleep_time_seconds != int(sleep_time_seconds):
+            raise ValueError('invalid sleep_time_seconds, must be integer \
+                value between 0 and 86399 inclusive')
+        else:
+            sleep_time = str(
```

Comment:
Okay, so I'm failing because I'm redefining the type of ```sleep_time``` from ```unicode``` to ```str```. This makes no sense to me, as ```sleep_time``` was previously undefined. Help please.